assets/src/edit-story/components/panels/design/preset/colorPreset/colorPresetActions.js

```diff
@@ -70,35 +129,109 @@ function ColorPresetActions({ color }) {
+          ? currentStoryStyles
+          : globalStoryStyles;
+        if (findMatchingColor(toAdd, currentStyles, isText)) {
           return;
         }
-
+        const newProps = showLocalColors
+          ? {
+              currentStoryStyles: {
+                ...currentStoryStyles,
+                colors: [...localColors, toAdd],
+              },
+            }
+          : {
+              globalStoryStyles: {
+                ...globalStoryStyles,
+                colors: [...globalColors, toAdd],
+              },
+            };
         updateStory({
-          properties: {
-            globalStoryStyles: {
-              ...globalStoryStyles,
-              colors: [...colors, toAdd],
-            },
-          },
+          properties: newProps,
         });
       }
     },
-    [globalStoryStyles, isText, colors, updateStory]
+    [
+      currentStoryStyles,
+      globalStoryStyles,
+      localColors,
+      showLocalColors,
+      isText,
+      globalColors,
+      updateStory,
+    ]
   );
 
+  const options = [
+    {
+      id: LOCAL,
+      name: __('Current colors', 'web-stories'),
+    },
+    {
+      id: GLOBAL,
+      name: __('Global colors', 'web-stories'),
+    },
+  ];
+
+  const savedColors = showLocalColors ? localColors : globalColors;
   return (
     <ActionsWrapper>
-      <AddColorPreset onClick={() => handleAddColorPreset(color)}>
-        {__('+ Add to Color Preset', 'web-stories')}
-      </AddColorPreset>
+      <HeaderRow>
+        <DropDownWrapper>
+          <AdvancedDropDown
+            options={options}
+            displayInContent={true}
+            hasSearch={false}
+            onChange={({ id }) => setShowLocalColors(id === LOCAL)}
+            selectedId={showLocalColors ? LOCAL : GLOBAL}
+            aria-label={__('Select color type', 'web-stories')}
+          />
+        </DropDownWrapper>
+        <ButtonWrapper>
+          <AddColorPreset
+            aria-label={__('Add color', 'web-stories')}
+            onClick={() => handleAddColorPreset(color)}
+          >
+            <Add />
+          </AddColorPreset>
+        </ButtonWrapper>
+      </HeaderRow>
+      <ColorsWrapper hasColors={savedColors.length > 0}>
+        {savedColors.length > 0 && (
+          <ColorGroup
+            isEditMode={false}
+            isLocal={showLocalColors}
+            colors={savedColors}
+            handleClick={applyStyle}
+            displayAdd={false}
+            colorGap={COLOR_GAP}
+          />
+        )}
+        {!savedColors.length && (
+          <CtaWrapper>
+            <TranslateWithMarkup
+              mapping={{
+                i: <Add width={12} height={12} />,
+                arrow: <Curve width={32} />,
+              }}
+            >
+              {__(
+                'Click <i></i> to save a color. <arrow></arrow>',
+                'web-stories'
+              )}
+            </TranslateWithMarkup>
+          </CtaWrapper>
```

Comment:
This doesn't seem to have the correct height. From the designs, the height of the panel should be identical when empty or when having content, but isn't currently - the empty state is shorter (should probably have some extra spacing top and bottom).

Design:
![Screen Shot 2021-02-02 at 13 28 47](https://user-images.githubusercontent.com/637548/106645890-fa8def80-655a-11eb-83df-b06e167741ed.png)

| Empty | Non-empty |
|-|-|
| ![Screen Shot 2021-02-02 at 13 31 29](https://user-images.githubusercontent.com/637548/106645992-1db89f00-655b-11eb-9cf6-84f010b800a0.png) | ![Screen Shot 2021-02-02 at 13 31 25](https://user-images.githubusercontent.com/637548/106645995-1e513580-655b-11eb-800c-ae7a96d604df.png) |

You can also see the slight misalignment of the arrow in the empty design - it should start in the middle of the text and have some space to the text, but it starts aligned with the bottom of the text and has no spacing.